DANS-KNAW/dd-workflow-step-vault-metadata
# DD-1313 use system metadata key

Fixes DD-1313: use system metadata key

# Description of changes
Use a configurable key to edit the dansDataVaultMetadata

# How to test

First setup a server (VM like dev_dataversenl) with the Dataverse code that is implementing the System metadata feature. Code is in PR: https://github.com/GlobalDataverseCommunityConsortium/dataverse/tree/DANS/system_metadata
Make sure it has the `dd-vault-metadata` installed and configured as Pre-publish workflow. 

Then build both the `dans-dataverse-client-lib` and the `dd-vault-metadata` with the fixes for DD-1313. 
Resulting in `dans-dataverse-client-lib-0.20.1-SNAPSHOT.jar` and `dd-vault-metadata-0.5.1-SNAPSHOT.jar`. 
Move the jar files to the `/opt/dans.knaw.nl/dd-vault-metadata/`, `bin` and `lib` subdirectories. 
Rename `dans-dataverse-client-lib-0.20.1-SNAPSHOT.jar` to `dans-dataverse-client-lib-0.20.0.jar`. 
Change the link for dd-vault-metadata.jar; `sudo ln -f -s dd-vault-metadata-0.5.1-SNAPSHOT.jar dd-vault-metadata.jar`. 
Note: might be simpler to install the generated RPM. 

Change the dd-vault-metadata service config; edit `/etc/opt/dans.knaw.nl/dd-vault-metadata/config.yml`. 
Add 
```	
	vaultMetadataKey: "1234"
```

And for sake of testing also change the loglevel to DEBUG there. 
Restart the service: `sudo systemctl start dd-vault-metadata`. 

Configure Dataverse to have a key on the `dansDataVaultMetadata` metadata block, using `asadmin`. 
Create a password alias with asadmin:
```
     echo "AS_ADMIN_ALIASPASSWORD=1234" > /tmp/key.txt
     ./asadmin create-password-alias --passwordfile /tmp/key.txt dataverse.metadata.block-system-metadata-keys.dansDataVaultMetadata
     rm /tmp/key.txt
```
!Note use `update-password-alias` it the key needs changing

Then add the key to the options
```
     ./asadmin create-jvm-options "\-Ddataverse.metadata.block-system-metadata-keys.dansDataVaultMetadata=\${ALIAS=dataverse.metadata.block-system-metadata-keys.dansDataVaultMetadata}"
Don't forget to restart Payara to have the jvm-options changes take effect!
```
 
# Related PRs

(Add links)

* https://github.com/DANS-KNAW/dans-dataverse-client-lib/pull/40

# Notify

@DANS-KNAW/dataversedans
